### ISSUE TITLE
rest-api endpoint to query min required weight for a certain date or date range? 

### ISSUE THREAD

**Issue by nsteinme** [cortex-lab/alyx#302]
We'd like to know the historical weight versus min required weight. For this two changes: 
1) refactor the water req code just a little bit to take an input date rather than assuming today, and calculate water req as it should have looked on that date (i.e. considering weighings up to then etc). 

2) Have a rest api endpoint where you can give a subject and date range and get the weight and min weight for all those dates. Ideally also compiles and returns the amount of water and amount of hydrogel given on all those dates as well, for simplicity.

**Comment by nsteinme** [comment]
related to #257

**Comment by nsteinme** [comment]
@laurax452

**Comment by nsteinme** [comment]
I'm removing "priority low" - it isn't exactly high priority but Laura and Miles think it would be very useful to see how weight varies relative to the 80% line, and how much water the mouse was getting relative to min. I have a little matlab gui that can display these things easily but it has to wait on this issue unless I were to re-implement the formulas in matlab, which we'd like to avoid...

**Comment by rossant** [comment]
(1) should be done in the PR. Could you make a full spec of what you want with (2) with possibly an example of a query and the corresponding JSON result?

**Comment by nsteinme** [comment]
Ok, let's say there's an endpoint called "water_requirements", so I query:
alyx.cortexlab.net/water_requirements?subject=mySubjName&start_date=2017-05-16&end_date=2017-05-18

Then I get back (using not-json here, but you'll get the idea):
- subject = 'mySubjName'
- records
  [-date='2017-05-16'
   -measured_weight=12.2
   -min_weight=12.0
   -water_given = 0.8
   -hydrogel_given = 0.0
   -min_water_req = 0.76
  ], 
  [-date='2017-05-17'
etc

"min_weight" here would be the 70% value - if it seems more sensible to you it could instead return "expected_weight" - either is fine, either can be calculated from the other. 

It may also be useful to query the formula directly, e.g. alyx.cortexlab.net/weight_formula?measured_weight=15.8&reference_weight=14.6&age=79&implant_weight=1.2&gender=male
But let's forget that for now unless someone needs it, I guess.

**Comment by rossant** [comment]
[Done on alyx-dev](https://alyx-dev.cortexlab.net/water-requirement/Theiler?start_date=2017-01-01&end_date=2017-06-01) ! Feel free to open a new issue if there are any changes to make.
